### PR TITLE
Adds tacticool helmet lights, adds siren helmet as premium item

### LINKED ISSUE #9279: Tacticool helmet lights for sec helmets.
helmet light
Icons for a security helmet light attachment. Comes in tan for tacticool desert camo.
Download link for sprites: http://s000.tinyupload.com/?file_id=02685895708682422056
Edit: Now also comes in goofy blinking siren version as well.
constable

Maybe have them bought from those vendors in cargo that noone uses and attached/detached with a screwdriver?

### LINKED ISSUE #9462: Feature request: Flashlight eyes
@9600bauds please

### PR DESCRIPTION
**Pull request by 9600bauds**
You may now attach flashlights to security helmets, the HoS' helmet, the Warden's helmet, and SWAT helmets. To unattach the flashlight, use a screwdriver.
New flashlight attachment sprites courtesy of Blithering (see #9279) (closes #9279) (closes #9462).
Added siren helmets as a premium item from Sec vending machines. When used, they WOOP WOOP due to lack of better sound: 0:00 in https://www.youtube.com/watch?v=_BxBs4f4RIU

I had to make Sec helmets into a subtype, did so in such a way that other helmets could easily get new attachments if sprites are provided in the future. Modifies the maps to accomplish this

92% tested

Pictures:
helmet light
mspaint_2016-04-12_01-58-16

### CODE REVIEW

File: code/modules/clothing/head/misc_special.dm [hunk 1]
```diff
@@ -5,6 +5,7 @@
  *		Ushanka
  *		Pumpkin head
  *		Kitty ears
+ *		Butt
```
Comment (9600bauds) [inline code/modules/clothing/head/misc_special.dm:8]: important

**Comment by Probe1** [comment]
woop
woop

dats dasound o da police

**Comment by 9600bauds** [comment]
For the longest time I misheard the lyrics and thought they said "that's the sound that I miss" and it made sense to me because I never heard the rest of the song

**Comment by Probe1** [comment]
been there, brain will remind me of it when I try to sleep.

**Comment by Hoshi-chan** [comment]
How is that for lack of a better sound?

You couldn't have picked a better sound.

**Comment by Blithering** [comment]
Looks good, how much of the song is played when activated?

**Comment by Probe1** [comment]
about 8 10ths of a second

**Comment by 9600bauds** [comment]
Literally just the WOOP WOOP

**Comment by liltiptop** [comment]
Dank
security

**Comment by Shadowmech88** [comment]
You say it plays the sound "when used." Does that mean you have to take the helmet off and click it in your hand in order to play the sound?
I'd prefer if it was a UI button or something like hardsuit helmet lights.

**Comment by 9600bauds** [comment]
It does have a UI button as well

**Comment by Blithering** [comment]
What other helmet types need the attachment? I'm happy to provide sprites for them.

**Comment by Brolaire** [comment]
HoS armor patch 😂👌

**Comment by Blithering** [comment]
@Brolaire 
stronger
HoS with dermal armour: 'I am growing stronger.'

**Comment by 9600bauds** [comment]
@Blithering If you feel like spriting more hats, I think maybe the Warden's and HoS' hat could use a flashlight attachment. The SWAT helmet is essentially the same sprite as the default helmet, too.

**Comment by Blithering** [comment]
@9600bauds  http://s000.tinyupload.com/index.php?file_id=17939740674329619515 There you go, let me know if any others spring to mind.

**Comment by Brolaire** [comment]
can the flashlight eyes be a real thing?

**Comment by Intigracy** [comment]
Why don't you just make the flashlight an overlay?

**Comment by 9600bauds** [comment]
I am much too lazy and it wasn't sprited like so

**Comment by 9600bauds** [comment]
Added @Blithering's FANTASTIC sprites, including the HoS' hat, SWAT helmet, and the Warden's cap. Also added the tactical light as an actual item, it's actually just a flashlight reskin lol
Only one small detail though: If you can sprite something better for ON sprite of the tactical light than this monstrosity I made then it would be appreciated
dreammaker_2016-04-12_01-42-46

**Comment by 9600bauds** [comment]
Of course the glowing eyes can be a real thing
mspaint_2016-04-12_01-48-36

**Comment by Shadowmech88** [comment]
It'd be neat to be able to attach the tactical light to children of `/obj/item/weapon/gun`.
Maybe make it shine brighter than a flashlight but only in the direction you're facing.

**Comment by 9600bauds** [comment]
Would require a ton of spritework, and probably wouldn't be too ez to make the directional light actually look good

**Comment by Shadowmech88** [comment]
You could just not make the sprite show up on the gun.

**Comment by 9600bauds** [comment]
Would be even lazier than my standards, and probably wouldn't be too ez to make the directional light actually look good

File: code/modules/clothing/head/helmet.dm [hunk 2]
```diff
@@ -1,20 +1,37 @@
...
+	var/spamcheck = 0
+
+/obj/item/clothing/head/helmet/siren/New()
+	..()
+	set_light(4)
+
+/obj/item/clothing/head/helmet/siren/attack_self(mob/user)
+	if(spamcheck)
+		return
+	playsound(get_turf(src), 'sound/voice/woopwoop.ogg', 100, 1, vary = 0)
```
Comment (9600bauds) [inline code/modules/clothing/head/helmet.dm:25]: Vary pitch y/n?

Comment (Shadowmech88) [inline code/modules/clothing/head/helmet.dm:25]: I don't think so. Police sirens are supposed to be a constant pitch, right?

Comment (9600bauds) [inline code/modules/clothing/head/helmet.dm:25]: I suspect this is essentially going to be a helmet-mounted bike horn that coincidentally happens to WOOP WOOP
there will actually be very little of a siren in the uses people will give it

Comment (Intigracy) [inline code/modules/clothing/head/helmet.dm:25]: y

Comment (ihadtoregisterforthis) [inline code/modules/clothing/head/helmet.dm:25]: n

**Comment by ihadtoregisterforthis** [comment]
Constant WOOP WOOP is better than _everything between_ "weep weep" _and_ "bwhup bwhup".
Even more if it's a voice clip.

**Comment by SysadminJeroen** [comment]
Merge when?

**Comment by 9600bauds** [comment]
when I code it

**Comment by Brolaire** [comment]
will this include tacticool eyes or is that too silly?

**Comment by 9600bauds** [comment]
See above

File: code/modules/clothing/head/helmet.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/modules/clothing/head/helmet.dm:25]: n, it'll sound weird as fuck

**Comment by Blithering** [comment]
http://s000.tinyupload.com/?file_id=36221188306375178146 Got the on state for the light, but the one you made is fine if you don't want the bother of re-downloading.

**Comment by 9600bauds** [comment]
ty4spriten

**Comment by Blithering** [comment]
No worries, thanks for coding it all.

**Comment by 9600bauds** [comment]
I just realized we missed the riot helmets
Other than that this is reddy 4 merge

**Comment by Blithering** [comment]
http://s000.tinyupload.com/index.php?file_id=08080124128622561331 Riot helmet added, also I realised that the regular helmet's light swapped sides when facing north, so that's fixed.

**Comment by 9600bauds** [comment]
:clap::clap::clap::clap:

**Comment by liltiptop** [comment]
Awesome. But where will we get the tacticool lights? Sec vendors?

**Comment by 9600bauds** [comment]
The sec helmets spawn with a tacticool light already mounted, and there are one on the Warden's and HoS' locker. They're literally just a flashlight reskin though, you can use regular flashlights too

**Comment by Brolaire** [comment]
tfw no flashlight eyes
I might cry now, but i'll get over it

**Comment by 9600bauds** [comment]
the flashlight eyes are in
this is the third time I tell you that

**Comment by Shadowmech88** [comment]
I like the new additions, but I really wish someone would add the flashlight eyes.

**Comment by Intigracy** [comment]
Bauds how come you won't add the flashlight eyes?

You're so mean.

**Comment by 9600bauds** [comment]
kcuf

**Comment by SysadminJeroen** [comment]
Merge when?

**Comment by Intigracy** [comment]
when he adds flashlight eyes :^)

**Comment by SysadminJeroen** [comment]
Just merge the PR already.

**Comment by Intigracy** [comment]
I can't lmao.

**Comment by Probe1** [comment]
image

File: code/modules/clothing/head/helmet.dm [hunk 3]
```diff
@@ -1,51 +1,50 @@
...
+	desc = "For the officer that's off patrolling all the nation."
+	icon_state = "helmetgoofy" //Sprites courtesy of Blithering
+	light_power = 2.5
+	light_color = LIGHT_COLOR_RED
+	action_button_name = "Activate Siren"
+	var/spamcheck = 0
+
+/obj/item/clothing/head/helmet/siren/New()
+	..()
+	set_light(4)
```
Comment (clusterfack) [inline code/modules/clothing/head/helmet.dm]: So this just has a light 24/7?

Comment (PJB3005) [inline code/modules/clothing/head/helmet.dm]: Side note: if you're gonna do that just set `light_range` on the type directly.

Comment (9600bauds) [inline code/modules/clothing/head/helmet.dm]: Ye, there are no off sprites. Besides, who would ever want to turn it off?

Comment (9600bauds) [inline code/modules/clothing/head/helmet.dm]: PJB was right with the light_range thing

**Comment by 9600bauds** [comment]
ty clusterman of optimization

**Comment by SysadminJeroen** [comment]
IT MERGED!!

**Comment by Blithering** [comment]
Its happening.

File: code/modules/clothing/head/helmet.dm [hunk 4]
```diff
@@ -1,51 +1,47 @@
...
+	light_range = 4
+	light_color = LIGHT_COLOR_RED
+	action_button_name = "Activate Siren"
+	var/spamcheck = 0
+
+/obj/item/clothing/head/helmet/siren/attack_self(mob/user)
+	if(spamcheck)
+		return
+	playsound(get_turf(src), 'sound/voice/woopwoop.ogg', 100, 1, vary = 0)
+	user.show_message("<span class='warning'>[user]'s [name] rasps, \"WOOP WOOP!\"</span>",1)
```
Comment (Intigracy) [inline code/modules/clothing/head/helmet.dm:26]: Should have been visible_message fam.

show_message is the functional equivilant of to_chat

Comment (9600bauds) [inline code/modules/clothing/head/helmet.dm:26]: shit

Comment (Intigracy) [inline code/modules/clothing/head/helmet.dm:26]: iirc the usage is visible_message("message", get_turf(src))

**Pull request merged**